src/platform/Linux/DnssdImpl.cpp

```diff
@@ -505,9 +505,13 @@ CHIP_ERROR MdnsAvahi::PublishService(const DnssdService & service)
     {
         avahi_string_list_free(text);
     }
-    if (error != CHIP_NO_ERROR)
+    if (CHIP_NO_ERROR == error)
     {
-        ChipLogError(DeviceLayer, "Avahi publish service failed: %" CHIP_ERROR_FORMAT, error.Format());
+        callback(context, type.c_str(), CHIP_NO_ERROR);
```

Comment:
I think, technically, the result is reported asynchronously in `HandleGroupState`.

Reply:
I think you are correct. Sadly Avahi is a bit painful here.

My understanding is that in  the `AVAHI_ENTRY_GROUP_ESTABLISHED` case of `HandleGroupState` one needs to retrieve the published type.

Again in my understanding the `Avahi` way of getting the type in the callback is to use the `void * userdata` pointer that the current code uses to pass a pointer to the current `MdnsAvahi` instance (basically passing `this` to `avahi_entry_group_new`).

And because that's fun, this code is called from an other callback `HandleClientState` which also uses the userdata to pass a pointer to the current instance. And this one is initialised from `MdnsAvahi::Init` that does not have clue about the published type at that point in time.

I have added a comment to the code. Ideally one would fix it, but I don't know if anybody is really using this code at that point in time since the current default for Linux is to use `minimal` and last time I tried to turned it on, CI was complaining about various tsan failures...